rust/src/storage/file/file_backend/page_cached_file.rs

```diff
@@ -107,55 +106,35 @@ impl<F: FileBackend, const D: bool> InnerPageCachedFile<F, D> {
             self.file
                 .write_all_at(&self.page, self.page_index * Page::SIZE as u64)?;
         }
-        self.file.flush()?;
-        self.set_len(self.file_len)
+        self.file.flush()
     }
 
     /// See [`FileBackend::len`].
     fn len(&self) -> BTResult<u64, std::io::Error> {
         Ok(self.file_len)
     }
 
-    /// See [`FileBackend::set_len`].
-    fn set_len(&mut self, len: u64) -> BTResult<(), std::io::Error> {
-        self.file_len = len;
-        self.file.set_len(len)
-    }
-
     /// Load the page containing the given offset into memory, flushing the current page if dirty.
     /// If the offset is already within the currently loaded page, this is a no-op.
     fn change_page(&mut self, offset: u64) -> BTResult<(), std::io::Error> {
         if offset < self.page_index * Page::SIZE as u64
             || offset >= (self.page_index + 1) * Page::SIZE as u64
         {
```

Comment:
```suggestion
        if offset >= self.page_index * Page::SIZE as u64
            && offset < (self.page_index + 1) * Page::SIZE as u64
        {
            // Page is already loaded.
            return Ok(());
        }
```

While we're here, we could get rid of the indentation in this function.